gobblin-runtime/src/main/java/gobblin/runtime/api/JobCatalog.java

```diff
@@ -24,7 +24,10 @@
   /** Returns an immutable {@link Collection} of {@link JobSpec}s that are known to the catalog. */
   Collection<JobSpec> getJobs();
 
-  /** Get a {@link JobSpec} by uri. */
+  /**
+   * Get a {@link JobSpec} by uri.
+   * @throws JobSpecNotFoundException if no such JobSpec exists
+   **/
   JobSpec getJobSpec(URI uri);
```

Comment:
My concern is if API users are not aware of the exception, and the exception does show up (eg a race condition where a spec has just been deleted but the notification has not been processed yet), the application will crash. If it is declared, user will be aware of it.